File: app/src/main/java/ru/coffeeplanter/androidlogin/presentation/fragments/loggedin/LoggedInPresenterImpl.java
package ru.coffeeplanter.androidlogin.presentation.fragments.loggedin;

import ru.coffeeplanter.androidlogin.domain.loggedin.LoggedInInteractor;
import ru.coffeeplanter.androidlogin.domain.loggedin.LoggedInInteractorImpl;

@SuppressWarnings("SpellCheckingInspection")
class LoggedInPresenterImpl implements LoggedInPresenter, LoggedInInteractor.OnUserForgottenListener {

    private LoggedInView loggedInView;
    private LoggedInInteractor interactor;

    LoggedInPresenterImpl(LoggedInView loggedInView) {
        this.loggedInView = loggedInView;
        interactor = new LoggedInInteractorImpl(this);
    }

    @Override
    public void onFragmentCreateView() {
        if (loggedInView != null) {
            loggedInView.getDataFromFragmentArguments();
            loggedInView.setGreetingMessage();
        }
    }

    @Override
    public void exitButtonClicked() {
        if (loggedInView != null) {
            loggedInView.closeApp();
        }
    }

    @Override
    public void forgetUserButtonClicked() {
        interactor.forgetUser();
    }

    @Override
    public void onFragmentDestroy() {
        loggedInView = null;
    }

    @Override
    public void onUserForget() {
        if (loggedInView != null) {
            loggedInView.navigateToLoginFragment();
        }
    }

}

File: app/src/main/java/ru/coffeeplanter/androidlogin/presentation/fragments/loggedin/LoggedInPresenterImpl.kt
package ru.coffeeplanter.androidlogin.presentation.fragments.loggedin

import ru.coffeeplanter.androidlogin.domain.loggedin.LoggedInInteractor
import ru.coffeeplanter.androidlogin.domain.loggedin.LoggedInInteractorImpl

internal class LoggedInPresenterImpl(private var loggedInView: LoggedInView?) : LoggedInPresenter, LoggedInInteractor.OnUserForgottenListener {
    private val interactor: LoggedInInteractor

    init {
        interactor = LoggedInInteractorImpl(this)
    }

    override fun onFragmentCreateView() {
        if (loggedInView != null) {
            loggedInView!!.getDataFromFragmentArguments()
            loggedInView!!.setGreetingMessage()
        }
    }

    override fun exitButtonClicked() {
        if (loggedInView != null) {
            loggedInView!!.closeApp()
        }
    }

    override fun forgetUserButtonClicked() {
        interactor.forgetUser()
    }

    override fun onFragmentDestroy() {
        loggedInView = null
    }

    override fun onUserForget() {
        if (loggedInView != null) {
            loggedInView!!.navigateToLoginFragment()
        }
    }

}
